game/gameboard.py

```diff
@@ -228,30 +225,39 @@ def _check_columns_for_winner(self):
     def _check_diagonals_for_winner(self):
         """Check if there's a winner diagonal-wise"""
         for i in range(-2, 4):
-            if self._winner_in_array(self._matrix.diagonal(i)) or self._winner_in_array(numpy.flipud(self._matrix).diagonal(i)):
+            if self._winner_in_array(self._matrix.diagonal(i)) or \
+               self._winner_in_array(numpy.flipud(self._matrix).diagonal(i)):
                 return True
         return False
 
     def _validate_matrix(self, matrix_to_test):
         """Checks if an input matrix have valid characters and structure
 
         Args:
-            matrix_to_test: numpy array with dtype=object
+            matrix_to_test: list
         Returns
             a boolean value
         """
-        valid_entries = (None, 'blue', 'red')
+        matrix_to_test = numpy.asarray(matrix_to_test)
         if matrix_to_test.shape != (6,7):
             return False
         for row in matrix_to_test:
             for entry in row:
```

Comment:
You can improve this loop with the use of `any()`. See https://docs.python.org/3/library/functions.html#any and http://stackoverflow.com/questions/1342601/pythonic-way-of-checking-if-a-condition-holds-for-any-element-of-a-list